### PR TITLE
Maint/6.0.x/merge up from 5.5.x

### PR DESCRIPTION
**Pull request by melissa**

### CODE REVIEW

**Comment by melissa** [comment]
jenkins please test this with servertests

File: lib/puppet/util/connection.rb [hunk 1]
```diff
@@ -22,15 +22,17 @@ module Connection
     # @return [String] the name of the server for use in the request
     def self.determine_server(setting)
       if setting && setting != :server && Puppet.settings.set_by_config?(setting)
+        Puppet.debug _("Selected server from the %{setting} setting: %{server}") % {setting: setting, server: Puppet.settings[setting]}
         Puppet[setting]
       else
         server = Puppet.lookup(:server) do
           if primary_server = Puppet.settings[:server_list][0]
-            Puppet.debug "Dynamically-bound server lookup failed; using first entry"
```
Comment (melissa) [inline lib/puppet/util/connection.rb]: @joshcooper Charlie's commit to change `Puppet.debug` to `debug_once` https://github.com/puppetlabs/puppet/commit/179f3e8e5d0cc059ccc4e77a96e1d828b716015d is in a later branch

File: lib/puppet/util/connection.rb [hunk 2]
```diff
@@ -51,19 +53,26 @@ def self.determine_server(setting)
...
+              primary_server[1]
+            else
+              #TRANSLATORS 'masterport' is the name of a setting and should not be translated
+              Puppet.debug _("Dynamically-bound port lookup failed; falling back to `masterport` setting: %{port}") % {port: Puppet.settings[:masterport]}
+              Puppet.settings[:masterport]
+            end
           else
             port_setting ||= :masterport
-            Puppet.debug "Dynamically-bound port lookup failed; falling back to #{port_setting} setting"
+            Puppet.debug _("Dynamically-bound port lookup failed; falling back to %{setting} setting: %{port}") % {setting: port_setting, port: Puppet.settings[setting]}
```
Comment (joshcooper) [inline lib/puppet/util/connection.rb]: does this need to be `port: Puppet.settings[port_setting]`?

Comment (melissa) [reply lib/puppet/util/connection.rb]: ah shoot, yes, good catch, thank you

**Comment by gimmyxd** [comment]
@melissa @joshcooper could this merge-up be responsible for `tests.server_list_setting.rb` failures?

https://jenkins-platform.delivery.puppetlabs.net/job/platform_puppet-agent_puppet-agent-integration-suite_daily-6.0.x/165/

https://jenkins-platform.delivery.puppetlabs.net/job/platform_puppet-agent_puppet-agent-integration-suite_daily-6.0.x/RMM_COMPONENT_TO_TEST_NAME=puppet,SLAVE_LABEL=beaker,TEST_TARGET=fedora28-64a/165/testReport/junit/(root)/tests/server_list_setting_rb/

**Comment by melissa** [comment]
yeah, most likely. I'm working on a fix now

**Pull request merged**